### DISCUSSION TITLE
How to connect device when ble already connected &paried to raspberrypi

### DISCUSSION THREAD

**Discussion by SOYOJOE** [hbldh/bleak#1087] [Q&A] [answered]
I have a ble keyboord,when the keyboard already paried with pi,it can input normolly,
then ,i use bleck with keyboard 's mac to connect to keyboard,it is failed,so what should do,
do it means it must use adv to connect ble device?

**Comment by dlech** [comment, marked as answer]
There is a feature request for this at #367.

On Linux, you may be able to hack around this by creating a `BLEDevice` object the the correct address and details and pass that to the `BleakClient` constructor instead of scanning.

**Reply by SOYOJOE** [reply]
thanks !
